cvprac/cvp_api.py

```diff
@@ -3206,4 +3242,470 @@ def workspace_build_status(self, workspace_id, build_id):
+        if 'approve' in cc_status['value']:
+            # For on-prem check the version as it is only supported from 2021.2.0+
+            if not self.clnt.is_cvaas:
+                if self.clnt.apiversion is None:
+                    self.get_cvp_info()
+                if self.clnt.apiversion >= 6.0:
+                    return self.clnt.get(url, timeout=self.request_timeout)
+                elif self.clnt.apiversion < 6.0:
+                    self.log.warning('change_control_approval_get_one: Change Control Resource APIs are supported'
+                                     ' from 2021.2.0 or newer.')
+                    return None
+            else:
+                return self.clnt.get(url, timeout=self.request_timeout)
+        else:
+            self.log.warning("change_control_approval_get_one: The change has not been approved yet."
+                             " A change has to be approved at least once for the 'approve' state to be populated.")
+            return None
+
+
+    def change_control_approval_get_all(self):
+        ''' Get state information for all Change Control Approvals using Resource APIs.
+            Supported versions: CVP 2021.2.0 or newer and CVaaS.
+
+            Returns:
+               response (dict): A dict that contains a list of all Change Control Approval Configs.
+        '''
+        url = '/api/resources/changecontrol/v1/ApproveConfig/all'
+        # For on-prem check the version as it is only supported from 2021.2.0+
+
+        if not self.clnt.is_cvaas:
+            if self.clnt.apiversion is None:
+                self.get_cvp_info()
+            if self.clnt.apiversion >= 6.0:
+                self.log.debug('v6 {}'.format(url))
+                response = self.clnt.get(url, timeout=self.request_timeout)
+                return response
+            elif self.clnt.apiversion < 6.0:
+                    self.log.warning('change_control_approval_get_all: Change Control Resource APIs are supported'
+                                     ' from 2021.2.0 or newer.')
+                    return None
+        else:
+            return self.clnt.get(url, timeout=self.request_timeout)
+
+    def change_control_approve(self, cc_id, notes="", approve=True):
+        ''' Approve/Unapprove a change control using Resource APIs.
+            Supported versions: CVP 2021.2.0 or newer and CVaaS.
+
+            Args:
+              cc_id (str): The ID of the change control.
+              notes (str): An optional approval note.
+              approve (bool): Set to True to approve a change and to False to unapprove a change. The default is True.
+        '''
+        url = '/api/resources/changecontrol/v1/ApproveConfig'
+        # For on-prem check the version as it is only supported from 2021.2.0+
+        # Since the get_change_control already checks this, no need to check it again
+        version = self.change_control_get_one(cc_id)['value']['change']['time']
+        payload = {
```

Comment:
I think it's still necessary to check the version as this function will just return `None`. Thus, when trying this function on 2021.1.1 version I got a `object not subscrictable` error message: 
```
clnt_hume.api.change_control_approve(cc_id='WzkdFQlBz', notes='salut', approve=True)
```
Result: 
```
TypeError                                 Traceback (most recent call last)
/var/folders/hd/3g0v07bx6vd9n3tn3ztkc5900000gn/T/ipykernel_20210/2122868010.py in <module>
----> 1 clnt_hume.api.change_control_approve(cc_id='WzkdFQlBz', notes='salut', approve=True)

~/Documents/tmp/cvprac/cvprac/cvp_api.py in change_control_approve(self, cc_id, notes, approve)
   3386         # For on-prem check the version as it is only supported from 2021.2.0+
   3387         # Since the get_change_control already checks this, no need to check it again
-> 3388         version = self.change_control_get_one(cc_id)['value']['change']['time']
   3389         payload = {
   3390             "key": {

TypeError: 'NoneType' object is not subscriptable
```

Reply:
I agree with this. The redundancy here is that the function is calling the get_one resource API and it's function also does the version check. Should we make the version an input param to the function instead of calling get_one to try to populate it?